lib/bgs/services/intent_to_file.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+# As a work of the United States Government, this project is in the
+# public domain within the United States.
+#
+# Additionally, we waive copyright and related rights in the work
+# worldwide through the CC0 1.0 Universal public domain dedication.
+
+module BGS
+  class IntentToFileWebService < BGS::Base
+    def bean_name
+      'IntentToFileWebServiceBean'
+    end
+  
+    def self.service_name
+      'intent_to_file'
+    end
+  
+    def find_intent_to_file_by_participant_id(participant_id)
+      response = request(
+        :find_intent_to_file_by_ptcpnt_id, 'ptcpntId': participant_id
+      )
+      response.body[:find_intent_to_file_by_ptcpnt_id_response][:return]
+    end
+
+    def insert_intent_to_file(options)
+      validate_required_keys(required_insert_intent_to_file_fields, options, __method__.to_s)
+  
+      response = request(
+          :insert_intent_to_file,
+          {
+            "intentToFileDTO": {
+              "itf_type_cd": options[:intent_to_file_type_code],
+              "ptcpntClaimantId": options[:participant_claimant_id],
+              "ptcpntVetId": options[:participant_vet_id],
+              "rcvdDt": options[:received_date],
+              "signtrInd": options[:signature_indicated],
+              "submtrAppIcnTypeCd": options[:submitter_application_icn_type_code]
+            }
+          },
+          options[:ssn]
+        )
+  
+      response.body[:insert_intent_to_file_response][:return]
+    end
+
+    def update_intent_to_file(options)
+      validate_required_keys(required_update_intent_to_file_fields, options, __method__.to_s)
+    
+      response = request(
+            :update_intent_to_file,
+            {
+              "intentToFileDTO": {
+                "createDt": options[:created_date],
+                "intentToFileId": options[:intent_to_file_id],
+                "itfStatusTypeCd": options[:intent_to_file_status_code],
+                "itf_type_cd": options[:intent_to_file_type_code],
+                "rcvdDt": options[:received_date],
+                "submtrAppIcnTypeCd": options[:submitter_application_icn_type_code],
+                "vetFileNbr": options[:veteran_file_number]
+              }
+            },
+            options[:ssn]
+          )
+    
+      response.body[:update_intent_to_file_response][:return]
+    end
+  
+    private
+  
+    def required_insert_intent_to_file_fields
+      %i[
+        intent_to_file_type_code
+        participant_claimant_id
+        participant_vet_id
+        received_date
+        signature_indicated
+        submitter_application_icn_type_code
```

Comment:
I wonder if we need to take this in as a field or if we should just set it from the application settings to the app name BGS gave us

Reply:
I would prefer we don't abbreviate the way that BGS does and explain what the values actually mean and represent so that there is no confusion due to abbreviated titles.

Reply:
I am not talking about abbreviation, the submitter application is the app who is making the request. Which would map to the application in the BGS configuration. In our case is VAgovAPI. What I am suggesting is that we dont allow this field to be passed because it could be dont incorrectly, rather we pull it from the configuration

Reply:
i paused here too, but, the more i think about it, this seems very reasonable